waku/node/peer_manager/peer_manager.nim

```diff
@@ -631,11 +632,14 @@ proc prunePeerStore*(pm: PeerManager) =
                                        capacity = capacity,
                                        pruned = pruned
 
-proc selectPeer*(pm: PeerManager, proto: string): Option[RemotePeerInfo] =
+proc selectPeer*(pm: PeerManager, proto: string, shard: Option[PubsubTopic] = none(PubsubTopic)): Option[RemotePeerInfo] =
```

Comment:
In the spirit of `shard` being the number encoded into the `pubsubTopic`, could we call this something like `shardedPubsubTopic`?

Reply:
> In the spirit of `shard` being the number encoded into the `pubsubTopic`, 

Is it? I always thought of shards as pubsub topic that conform to static or autoshard specs. The numbers are named cluster and index.

> could we call this something like `shardedPubsubTopic`?
The term `shardedPubsubTopic` has never been used before. I don't like adding more terminology.